src/components/structures/RoomView.tsx

```diff
@@ -587,18 +540,18 @@ export class RoomView extends React.Component<IRoomProps, IRoomState> {
             // The rest will be lost for now, until the aggregation API on the server
             // becomes available to fetch a whole thread
             if (!initialEvent) {
-                initialEvent = await fetchInitialEvent(this.context.client, roomId, initialEventId);
+                initialEvent = (await fetchInitialEvent(client, roomId!, initialEventId)) || undefined;
             }
 
             // If we have an initial event, we want to reset the event pixel offset to ensure it ends up
             // visible
-            newState.initialEventPixelOffset = null;
+            newState.initialEventPixelOffset = undefined;
 
             const thread = initialEvent?.getThread();
             if (thread && !initialEvent?.isThreadRoot) {
                 dis.dispatch<ShowThreadPayload>({
                     action: Action.ShowThread,
-                    rootEvent: thread.rootEvent,
+                    rootEvent: thread.rootEvent!, // isThreadRoot guarantees
```

Comment:
Logically yes. Technically it doesn't. Shouldn't `initialEvent` be the root event?

Reply:
I'm not familiar enough with the threads API to say. Maybe? There aren't any types here to make this work without some kind of assertion. I'll swap it, I was just trying to change as little as possible while adding context.